### ISSUE TITLE
Planning failure for group by query with field dereference

### ISSUE THREAD

**Issue by martint** [facebook/presto#3823]
Given a table with the following shape:

```
t (
   x array<row(key varchar, value bigint)>
)
```

This query:
```
SELECT x[1].key, count(1) 
FROM t 
GROUP BY 1 
ORDER BY 2
```

Fails with:
```
java.lang.IllegalArgumentException: No mapping for expression: "x"[1]."key"
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at com.facebook.presto.sql.planner.TranslationMap.get(TranslationMap.java:153)
	at com.facebook.presto.sql.planner.TranslationMap.get(TranslationMap.java:165)
	at com.facebook.presto.sql.planner.PlanBuilder.translate(PlanBuilder.java:63)
	at com.facebook.presto.sql.planner.QueryPlanner.aggregate(QueryPlanner.java:386)
	at com.facebook.presto.sql.planner.QueryPlanner.visitQuerySpecification(QueryPlanner.java:131)
	at com.facebook.presto.sql.planner.QueryPlanner.visitQuerySpecification(QueryPlanner.java:80)
	at com.facebook.presto.sql.tree.QuerySpecification.accept(QuerySpecification.java:98)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:22)
	at com.facebook.presto.sql.planner.RelationPlanner.visitQuerySpecification(RelationPlanner.java:394)
	at com.facebook.presto.sql.planner.RelationPlanner.visitQuerySpecification(RelationPlanner.java:94)
	at com.facebook.presto.sql.tree.QuerySpecification.accept(QuerySpecification.java:98)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:22)
	at com.facebook.presto.sql.planner.QueryPlanner.planQueryBody(QueryPlanner.java:151)
	at com.facebook.presto.sql.planner.QueryPlanner.visitQuery(QueryPlanner.java:107)
	at com.facebook.presto.sql.planner.QueryPlanner.visitQuery(QueryPlanner.java:80)
	at com.facebook.presto.sql.tree.Query.accept(Query.java:80)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:22)
	at com.facebook.presto.sql.planner.RelationPlanner.visitQuery(RelationPlanner.java:381)
	at com.facebook.presto.sql.planner.RelationPlanner.visitQuery(RelationPlanner.java:94)
	at com.facebook.presto.sql.tree.Query.accept(Query.java:80)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:22)
	at com.facebook.presto.sql.planner.LogicalPlanner.createRelationPlan(LogicalPlanner.java:204)
	at com.facebook.presto.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:89)
	at com.facebook.presto.execution.SqlQueryExecution.doAnalyzeQuery(SqlQueryExecution.java:266)
	at com.facebook.presto.execution.SqlQueryExecution.analyzeQuery(SqlQueryExecution.java:245)
	at com.facebook.presto.execution.SqlQueryExecution.start(SqlQueryExecution.java:209)
	at com.facebook.presto.execution.QueuedExecution.lambda$start$171(QueuedExecution.java:68)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

If `key` is no de-referenced, the query works fine. This works on version 0.123, so this could be related to the changes in db5ea9183412656f9cbf2e8a9302666ac323143d

**Comment by cberner** [comment]
@joy-yao was looking into this

**Comment by raghavsethi** [comment]
@joy-yao fixed this in #3830.
